src/__tests__/components/card-list.test.tsx

```diff
@@ -0,0 +1,53 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import CardList from '../../components/card-list/card-list';
+import type { Character } from '../../core/interfaces/interface';
+
+vi.mock('fetch');
+
+describe('CardList content', () => {
+  let response: Character[];
+  beforeEach(() => {
+    response = [
+      {
+        gender: 'Male',
+        id: 1,
+        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
+        name: 'Rick Sanchez',
+        species: 'Human',
+        status: 'Alive',
+      },
+      {
+        gender: 'Male',
+        id: 2,
+        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
+        name: 'Morty Smith',
+        species: 'Human',
+        status: 'Alive',
+      },
+      {
+        gender: 'Female',
+        id: 3,
+        image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg',
+        name: 'Summer Smith',
+        species: 'Human',
+        status: 'Alive',
+      },
+    ];
+  });
+
```

Comment:
```suggestion

const response: Character[] = [
      {
        gender: 'Male',
        id: 1,
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
        name: 'Rick Sanchez',
        species: 'Human',
        status: 'Alive',
      },
      {
        gender: 'Male',
        id: 2,
        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
        name: 'Morty Smith',
        species: 'Human',
        status: 'Alive',
      },
      {
        gender: 'Female',
        id: 3,
        image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg',
        name: 'Summer Smith',
        species: 'Human',
        status: 'Alive',
      },
    ];

describe('CardList content', () => {

```
You do not modify mocked data so better to put it out of describe function